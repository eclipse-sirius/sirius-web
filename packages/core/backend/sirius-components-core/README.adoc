= sirius-components-core

== Goal

This project is used to contain the API to manipulate core concepts that are not specific to the business of our application.

== Dependencies

- sirius-components-annotations

This project should not depend on anything else without a good reason to do so.