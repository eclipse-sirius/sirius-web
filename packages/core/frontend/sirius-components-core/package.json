{
  "name": "@eclipse-sirius/sirius-components-core",
  "version": "2024.5.0",
  "author": "Eclipse Sirius",
  "license": "EPL-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/eclipse-sirius/sirius-web"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "main": "./dist/sirius-components-core.umd.js",
  "module": "./dist/sirius-components-core.es.js",
  "types": "./dist/index.d.ts",
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "require": "./dist/sirius-components-core.umd.js",
      "import": "./dist/sirius-components-core.es.js"
    }
  },
  "scripts": {
    "start": "vite build --mode 'development' && tsc",
    "build": "vite build && tsc",
    "format": "prettier --write \"src/**/*.{js,ts,tsx,css}\"",
    "format-lint": "prettier --list-different \"src/**/*.{js,ts,tsx,css}\"",
    "test": "vitest --run --config vite.config.js",
    "coverage": "vitest run --config vite.config.js --coverage",
    "publish:local": "yalc push"
  },
  "peerDependencies": {
    "@apollo/client": "3.10.4",
    "@material-ui/core": "4.12.4",
    "@material-ui/icons": "4.11.3",
    "@xstate/react": "1.6.3",
    "graphql": "16.8.1",
    "react": "17.0.2",
    "xstate": "4.32.1"
  },
  "devDependencies": {
    "@apollo/client": "3.10.4",
    "@eclipse-sirius/sirius-components-tsconfig": "*",
    "@material-ui/core": "4.12.4",
    "@material-ui/icons": "4.11.3",
    "@types/react": "17.0.37",
    "@vitejs/plugin-react": "4.0.4",
    "@xstate/react": "1.6.3",
    "@vitest/coverage-v8": "0.34.2",
    "jsdom": "16.7.0",
    "graphql": "16.8.1",
    "prettier": "2.7.1",
    "react": "17.0.2",
    "rollup-plugin-peer-deps-external": "2.2.4",
    "xstate": "4.32.1",
    "typescript": "5.1.6",
    "vite": "4.4.9",
    "vitest": "0.34.2"
  }
}
