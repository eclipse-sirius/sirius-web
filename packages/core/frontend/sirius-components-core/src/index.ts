/*******************************************************************************
 * Copyright (c) 2022, 2024 Obeo and others.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
export * from './color/getCSSColor';
export * from './contexts/RepresentationPathContext';
export * from './contexts/RepresentationPathContext.types';
export * from './contexts/ServerContext';
export * from './contexts/ServerContext.types';
export * from './contexts/ToastContext';
export * from './contexts/ToastContext.types';
export * from './dataTransferTypes';
export { ExtensionProvider } from './extension/ExtensionProvider';
export { type ExtensionProviderProps } from './extension/ExtensionProvider.types';
export { ExtensionRegistry } from './extension/ExtensionRegistry';
export {
  type ComponentExtension,
  type ComponentExtensionPoint,
  type DataExtension,
  type DataExtensionPoint,
} from './extension/ExtensionRegistry.types';
export { type ExtensionRegistryMergeStrategy } from './extension/ExtensionRegistryMergeStrategy';
export { useComponent } from './extension/useComponent';
export { useComponents } from './extension/useComponents';
export { useData } from './extension/useData';
export * from './graphql/GQLTypes.types';
export * from './icon/IconOverlay';
export * from './materialui';
export * from './modals/confirmation/ConfirmationDialogContext';
export * from './modals/confirmation/ConfirmationDialogContext.types';
export * from './modals/confirmation/useConfirmationDialog';
export * from './modals/confirmation/useConfirmationDialog.types';
export * from './modals/share-representation/ShareRepresentationModal';
export * from './modals/share-representation/ShareRepresentationModal.types';
export * from './selection/SelectionContext';
export * from './selection/SelectionContext.types';
export * from './selection/useSelection';
export * from './selection/useSelection.types';
export * from './theme';
export * from './toast/MultiToast';
export * from './toast/Toast';
export * from './toast/useReporting';
export * from './workbench/Panels';
export * from './workbench/RepresentationContext';
export * from './workbench/RepresentationContext.types';
export * from './workbench/Workbench';
export * from './workbench/Workbench.types';
export * from './workbench/WorkbenchExtensionPoints';
