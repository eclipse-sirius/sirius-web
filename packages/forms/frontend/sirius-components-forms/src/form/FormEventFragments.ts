/*******************************************************************************
 * Copyright (c) 2021, 2024 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/

export const commonFields = `
  fragment commonFields on Widget {
    id
    __typename
    diagnostics {
      id
      kind
      message
    }
    hasHelpText
    readOnly
  }
`;

export const widgetFields = `
  ${commonFields}
  
  fragment textfieldFields on Textfield {
    label
    iconURL
    stringValue: value
    supportsCompletion
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment textareaFields on Textarea {
    label
    iconURL
    stringValue: value
    supportsCompletion
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment checkboxFields on Checkbox {
    label
    iconURL
    booleanValue: value
    style {
      color
      labelPlacement
    }
  }

  fragment selectFields on Select {
    label
    iconURL
    value
    options {
      id
      label
      iconURL
    }
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment multiSelectFields on MultiSelect {
    label
    iconURL
    values
    options {
      id
      label
      iconURL
    }
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment radioFields on Radio {
    label
    iconURL
    options {
      id
      label
      selected
    }
    style {
      color
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment listFields on List {
    label
    iconURL
    items {
      id
      label
      kind
      iconURL
      deletable
    }
    style {
      color
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment linkFields on Link {
    label
    iconURL
    url
    style {
      color
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment buttonFields on Button {
    label
    buttonLabel
    imageURL
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment splitButtonFields on SplitButton {
    label
    actions {
      ...commonFields
      ...buttonFields
    }
  }

  fragment toolbarActionFields on ToolbarAction {
    label
    buttonLabel
    imageURL
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment chartWidgetFields on ChartWidget {
    label
    chart {
      __typename
      ... on BarChart {
        metadata {
          label
        }
        entries {
          key
          value
        }
        style {
          barsColor
          fontSize
          italic
          bold
          underline
          strikeThrough
        }
        width
        height
      }
      ... on PieChart {
        metadata {
          label
        }
        entries {
          key
          value
        }
        style {
          colors
          strokeColor
          strokeWidth
          fontSize
          italic
          bold
          underline
          strikeThrough
        }
      }
    }
  } 

  fragment labelWidgetFields on LabelWidget {
    label
    stringValue: value
    style {
      color
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment treeWidgetFields on TreeWidget {
    label
    iconURL
    expandedNodesIds
    nodes {
      id
      parentId
      label
      kind
      iconURL
      endIconsURL
      selectable
      checkable
      value
    }
  }

  fragment imageFields on Image {
    label
    iconURL
    url
    maxWidth
  }

  fragment richTextFields on RichText {
    label
    iconURL
    stringValue: value
  }

  fragment sliderFields on Slider {
    label
    iconURL
    minValue
    maxValue
    currentValue
  }
  
  fragment dateTimeFields on DateTime {
    label
    iconURL
    stringValue
    type
    style {
      backgroundColor
      foregroundColor
      italic
      bold
    }
  }

  fragment widgetFields on Widget {
    ...commonFields
    ... on Textfield {
      ...textfieldFields
    }
    ... on Textarea {
      ...textareaFields
    }
    ... on Checkbox {
      ...checkboxFields
    }
    ... on Select {
      ...selectFields
    }
    ... on MultiSelect {
      ...multiSelectFields
    }
    ... on Radio {
      ...radioFields
    }
    ... on List {
      ...listFields
    }
    ... on Link {
      ...linkFields
    }
    ... on Button {
      ...buttonFields
    }
    ... on SplitButton {
      ...splitButtonFields
    }
    ... on ToolbarAction {
      ...toolbarActionFields
    }
    ... on LabelWidget {
      ...labelWidgetFields
    }
    ... on ChartWidget {
      ...chartWidgetFields
    }
    ... on TreeWidget {
      ...treeWidgetFields
    }
    ... on Image {
      ...imageFields
    }
    ... on RichText {
      ...richTextFields
    }
    ... on Slider {
      ...sliderFields
    }
    ... on DateTime {
      ...dateTimeFields
    }
  }
`;

export const flexboxContainerFields = `
  ${commonFields}
  ${widgetFields}
  fragment flexboxContainerFields on FlexboxContainer {
    ...commonFields
    label
    flexDirection
    flexWrap
    flexGrow
    children {
      ...widgetFields
      ... on FlexboxContainer {
        ...commonFields
        label
        flexDirection
        flexWrap
        flexGrow
        children {
          ...widgetFields
          ... on FlexboxContainer {
            ...commonFields
            label
            flexDirection
            flexWrap
            flexGrow
            children {
              ...widgetFields
            }
            borderStyle {
              color
              lineStyle
              size
              radius
            }
          }
        }
        borderStyle {
          color
          lineStyle
          size
          radius
        }
      }
    }
    borderStyle {
      color
      lineStyle
      size
      radius
    }
  }
`;

export const formRefreshedEventPayloadFragment = `
  ${widgetFields}
  ${flexboxContainerFields}
  fragment formRefreshedEventPayloadFragment on FormRefreshedEventPayload {
    id
    form {
      id
      pages {
        id
        label
        groups {
          id
          label
          displayMode
          toolbarActions {
            ...commonFields
            ...toolbarActionFields
          }
          borderStyle {
            color
            lineStyle
            size
            radius
          }
          widgets {
            ...widgetFields
            ... on FlexboxContainer {
              ...flexboxContainerFields
            }
          }
        }
        toolbarActions {
            ...commonFields
            ...toolbarActionFields
        }
      }
    }
  }
`;
