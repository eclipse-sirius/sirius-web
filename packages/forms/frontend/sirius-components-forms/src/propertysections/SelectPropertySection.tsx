/*******************************************************************************
 * Copyright (c) 2019, 2024 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import { gql, useMutation } from '@apollo/client';
import { IconOverlay, getCSSColor, useMultiToast } from '@eclipse-sirius/sirius-components-core';
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import { Theme, makeStyles } from '@material-ui/core/styles';
import { useEffect, useState } from 'react';
import { PropertySectionComponent, PropertySectionComponentProps } from '../form/Form.types';
import { GQLSelect } from '../form/FormEventFragments.types';
import { PropertySectionLabel } from './PropertySectionLabel';
import {
  GQLEditSelectMutationData,
  GQLEditSelectPayload,
  GQLErrorPayload,
  GQLSuccessPayload,
  GQLUpdateWidgetFocusMutationData,
  SelectStyleProps,
} from './SelectPropertySection.types';
import { getTextDecorationLineValue } from './getTextDecorationLineValue';

const useStyle = makeStyles<Theme, SelectStyleProps>((theme) => ({
  style: {
    backgroundColor: ({ backgroundColor }) => (backgroundColor ? getCSSColor(backgroundColor, theme) : null),
    color: ({ foregroundColor }) => (foregroundColor ? getCSSColor(foregroundColor, theme) : null),
    fontSize: ({ fontSize }) => (fontSize ? fontSize : null),
    fontStyle: ({ italic }) => (italic ? 'italic' : null),
    fontWeight: ({ bold }) => (bold ? 'bold' : null),
    textDecorationLine: ({ underline, strikeThrough }) => getTextDecorationLineValue(underline, strikeThrough),
  },
  iconRoot: {
    minWidth: theme.spacing(3),
  },
}));

export const editSelectMutation = gql`
  mutation editSelect($input: EditSelectInput!) {
    editSelect(input: $input) {
      __typename
      ... on ErrorPayload {
        messages {
          body
          level
        }
      }
      ... on SuccessPayload {
        messages {
          body
          level
        }
      }
    }
  }
`;

const updateWidgetFocusMutation = gql`
  mutation updateWidgetFocus($input: UpdateWidgetFocusInput!) {
    updateWidgetFocus(input: $input) {
      __typename
      ... on ErrorPayload {
        messages {
          body
          level
        }
      }
    }
  }
`;

const isErrorPayload = (payload: GQLEditSelectPayload): payload is GQLErrorPayload =>
  payload.__typename === 'ErrorPayload';
const isSuccessPayload = (payload: GQLEditSelectPayload): payload is GQLSuccessPayload =>
  payload.__typename === 'SuccessPayload';

export const SelectPropertySection: PropertySectionComponent<GQLSelect> = ({
  editingContextId,
  formId,
  widget,
  subscribers,
  readOnly,
}: PropertySectionComponentProps<GQLSelect>) => {
  const props: SelectStyleProps = {
    backgroundColor: widget.style?.backgroundColor ?? null,
    foregroundColor: widget.style?.foregroundColor ?? null,
    fontSize: widget.style?.fontSize ?? null,
    italic: widget.style?.italic ?? null,
    bold: widget.style?.bold ?? null,
    underline: widget.style?.underline ?? null,
    strikeThrough: widget.style?.strikeThrough ?? null,
  };
  const classes = useStyle(props);

  const [isFocused, setFocus] = useState(false);

  const [editSelect, { loading, error, data }] = useMutation<GQLEditSelectMutationData>(editSelectMutation);
  const onChange = (event) => {
    const newValue = event.target.value;
    const variables = {
      input: {
        id: crypto.randomUUID(),
        editingContextId,
        representationId: formId,
        selectId: widget.id,
        newValue,
      },
    };
    editSelect({ variables });
  };

  const { addErrorMessage, addMessages } = useMultiToast();

  useEffect(() => {
    if (!loading) {
      if (error) {
        addErrorMessage('An unexpected error has occurred, please refresh the page');
      }
      if (data) {
        const { editSelect } = data;
        if (isErrorPayload(editSelect) || isSuccessPayload(editSelect)) {
          addMessages(editSelect.messages);
        }
      }
    }
  }, [loading, error, data]);

  const [
    updateWidgetFocus,
    { loading: updateWidgetFocusLoading, data: updateWidgetFocusData, error: updateWidgetFocusError },
  ] = useMutation<GQLUpdateWidgetFocusMutationData>(updateWidgetFocusMutation);
  const sendUpdateWidgetFocus = (selected: boolean) => {
    const variables = {
      input: {
        id: crypto.randomUUID(),
        editingContextId,
        representationId: formId,
        widgetId: widget.id,
        selected,
      },
    };
    updateWidgetFocus({ variables });
  };

  useEffect(() => {
    if (!updateWidgetFocusLoading) {
      if (updateWidgetFocusError) {
        addErrorMessage('An unexpected error has occurred, please refresh the page');
      }
      if (updateWidgetFocusData) {
        const { updateWidgetFocus } = updateWidgetFocusData;
        if (isErrorPayload(updateWidgetFocus)) {
          addMessages(updateWidgetFocus.messages);
        }
      }
    }
  }, [updateWidgetFocusLoading, updateWidgetFocusData, updateWidgetFocusError]);

  const onFocus = () => {
    if (!isFocused) {
      setFocus(true);
      sendUpdateWidgetFocus(true);
    }
  };
  const onBlur = () => {
    setFocus(false);
    sendUpdateWidgetFocus(false);
  };

  return (
    <FormControl error={widget.diagnostics.length > 0}>
      <PropertySectionLabel
        editingContextId={editingContextId}
        formId={formId}
        widget={widget}
        subscribers={subscribers}
      />
      <Select
        value={widget.value || ''}
        onChange={onChange}
        displayEmpty
        onFocus={onFocus}
        onBlur={onBlur}
        fullWidth
        data-testid={widget.label}
        disabled={readOnly || widget.readOnly}
        inputProps={
          widget.style
            ? {
                className: classes.style,
              }
            : {}
        }>
        <MenuItem
          value=""
          classes={
            widget.style
              ? {
                  root: classes.style,
                }
              : {}
          }>
          <em>None</em>
        </MenuItem>
        {widget.options.map((option) => (
          <MenuItem
            value={option.id}
            key={option.id}
            classes={
              widget.style
                ? {
                    root: classes.style,
                  }
                : {}
            }>
            {option.iconURL.length > 0 && (
              <ListItemIcon className={classes.iconRoot}>
                <IconOverlay iconURL={option.iconURL} alt={option.label} />
              </ListItemIcon>
            )}

            {option.label}
          </MenuItem>
        ))}
      </Select>
      <FormHelperText>{widget.diagnostics[0]?.message}</FormHelperText>
    </FormControl>
  );
};
