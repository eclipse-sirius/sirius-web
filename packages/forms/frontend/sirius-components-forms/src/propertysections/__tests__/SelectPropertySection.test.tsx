/*******************************************************************************
 * Copyright (c) 2022, 2024 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import { MockedProvider } from '@apollo/client/testing';
import { MessageOptions, ToastContext, ToastContextValue } from '@eclipse-sirius/sirius-components-core';
import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { afterEach, expect, test, vi } from 'vitest';
import { GQLSelect } from '../../form/FormEventFragments.types';
import { SelectPropertySection } from '../SelectPropertySection';

crypto.randomUUID = vi.fn(() => '48be95fc-3422-45d3-b1f9-d590e847e9e1');

afterEach(() => cleanup());

const defaultSelect = {
  __typename: 'Select',
  id: 'selectId',
  label: 'SelectLabel',
  iconURL: [],
  hasHelpText: false,
  diagnostics: [],
  value: '',
  options: [],
  readOnly: false,
};

const selectWithStyle: GQLSelect = {
  ...defaultSelect,
  style: {
    backgroundColor: '#de1000',
    foregroundColor: '#fbb800',
    fontSize: 14,
    italic: false,
    bold: false,
    underline: false,
    strikeThrough: false,
  },
};

const selectWithEmptyStyle: GQLSelect = {
  ...defaultSelect,
  style: {
    backgroundColor: '',
    foregroundColor: '',
    fontSize: 14,
    italic: false,
    bold: false,
    underline: false,
    strikeThrough: false,
  },
};

const readOnlySelect: GQLSelect = {
  ...selectWithEmptyStyle,
  readOnly: true,
};

const mockEnqueue = vi.fn<[string, MessageOptions?], void>();

const toastContextMock: ToastContextValue = {
  enqueueSnackbar: mockEnqueue,
};

test('should render the select without style', () => {
  const { baseElement } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectPropertySection
          editingContextId="editingContextId"
          formId="formId"
          widget={selectWithEmptyStyle}
          readOnly={false}
        />
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(baseElement).toMatchSnapshot();
});

test('should render the select with style', () => {
  const { baseElement } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectPropertySection
          editingContextId="editingContextId"
          formId="formId"
          widget={selectWithStyle}
          readOnly={false}
        />
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(baseElement).toMatchSnapshot();
});

test('should render the select with empty style', async () => {
  const { baseElement } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectPropertySection
          editingContextId="editingContextId"
          formId="formId"
          widget={selectWithEmptyStyle}
          readOnly={false}
        />
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(baseElement).toMatchSnapshot();
});

test('should render the select with help hint', () => {
  const { baseElement } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectPropertySection
          editingContextId="editingContextId"
          formId="formId"
          widget={{ ...selectWithEmptyStyle, hasHelpText: true }}
          readOnly={false}
        />
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(baseElement).toMatchSnapshot();
});

test('should render a readOnly select from widget properties', () => {
  const { baseElement } = render(
    <MockedProvider>
      <ToastContext.Provider value={toastContextMock}>
        <SelectPropertySection
          editingContextId="editingContextId"
          formId="formId"
          widget={readOnlySelect}
          readOnly={false}
        />
      </ToastContext.Provider>
    </MockedProvider>
  );
  expect(baseElement).toMatchSnapshot();
});
