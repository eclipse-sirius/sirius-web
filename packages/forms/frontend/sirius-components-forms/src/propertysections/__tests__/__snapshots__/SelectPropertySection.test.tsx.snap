// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render a readOnly select from widget properties 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="css-1ccipwk-propertySectionLabel"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-t33jlf-typography css-ati914-MuiTypography-root"
        >
          SelectLabel
        </h6>
        <div
          class="css-vrp56w-decorators"
        />
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-u5kk7y-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        data-testid="SelectLabel"
      >
        <div
          aria-controls=":r5:"
          aria-disabled="true"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-standard Mui-disabled MuiInputBase-input MuiInput-input Mui-disabled css-ve6bqz-style css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          role="combobox"
        >
          <em>
            None
          </em>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          disabled=""
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard Mui-disabled css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      />
    </div>
  </div>
</body>
`;

exports[`should render the select with empty style 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="css-1ccipwk-propertySectionLabel"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-t33jlf-typography css-ati914-MuiTypography-root"
        >
          SelectLabel
        </h6>
        <div
          class="css-vrp56w-decorators"
        />
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-u5kk7y-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        data-testid="SelectLabel"
      >
        <div
          aria-controls=":r2:"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-ve6bqz-style css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          role="combobox"
          tabindex="0"
        >
          <em>
            None
          </em>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      />
    </div>
  </div>
</body>
`;

exports[`should render the select with help hint 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="css-1ccipwk-propertySectionLabel"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-t33jlf-typography css-ati914-MuiTypography-root"
        >
          SelectLabel
        </h6>
        <div
          class="css-vrp56w-decorators"
        >
          <svg
            aria-hidden="true"
            aria-label=""
            class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
            data-mui-internal-clone-element="true"
            data-testid="HelpOutlineOutlinedIcon"
            focusable="false"
            style="margin-left: 8px; font-size: 16px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-u5kk7y-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        data-testid="SelectLabel"
      >
        <div
          aria-controls=":r4:"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-ve6bqz-style css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          role="combobox"
          tabindex="0"
        >
          <em>
            None
          </em>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      />
    </div>
  </div>
</body>
`;

exports[`should render the select with style 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="css-1ccipwk-propertySectionLabel"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-t33jlf-typography css-ati914-MuiTypography-root"
        >
          SelectLabel
        </h6>
        <div
          class="css-vrp56w-decorators"
        />
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-u5kk7y-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        data-testid="SelectLabel"
      >
        <div
          aria-controls=":r1:"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-ijey5v-style css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          role="combobox"
          tabindex="0"
        >
          <em>
            None
          </em>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      />
    </div>
  </div>
</body>
`;

exports[`should render the select without style 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="css-1ccipwk-propertySectionLabel"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-t33jlf-typography css-ati914-MuiTypography-root"
        >
          SelectLabel
        </h6>
        <div
          class="css-vrp56w-decorators"
        />
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-u5kk7y-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        data-testid="SelectLabel"
      >
        <div
          aria-controls=":r0:"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-ve6bqz-style css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          role="combobox"
          tabindex="0"
        >
          <em>
            None
          </em>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      />
    </div>
  </div>
</body>
`;
